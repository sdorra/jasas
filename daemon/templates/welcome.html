{{ define "content" }}

  <div class="welcome">
    <p>Welcome Benutzer</p>
    <button class="btn-primary">Logout</button>
  </div>

{{ end }}
