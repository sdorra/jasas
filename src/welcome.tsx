import * as React from "react";

interface Props {
  username: string;
  logout(): void;
}

export class Welcome extends React.Component<Props, any> {

  constructor(props: Props) {
    super(props);
    this.handleLogoutClick = this.handleLogoutClick.bind(this);
  }

  handleLogoutClick(e: any) {
    e.preventDefault();
    this.props.logout();
  }

  render() {
    return (
      <div className="welcome">
        <p>Welcome {this.props.username}</p>
        <button onClick={this.handleLogoutClick}>Logout</button>
      </div>
    )
  }
}
