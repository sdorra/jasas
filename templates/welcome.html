{{ define "content" }}

  <div class="welcome">
    <p>Welcome {{.Username}}</p>
    <a class="btn-primary" href="logout">Logout</a>
  </div>

{{ end }}
