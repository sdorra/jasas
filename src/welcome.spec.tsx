import * as React from 'react';
import { create, ReactTestRendererJSON } from 'react-test-renderer';

import { Welcome } from './welcome';

describe('welcome', () => {

  it('should render text and button', () => {
    const logout = () => {};
    const tree = create(
      <Welcome username="tricia" logout={logout} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call logout, after button click', () => {
    let pressed: number = 0;
    const logout = () => {
      pressed++;
    };
    const tree = create(
      <Welcome username="tricia" logout={logout} />
    ).toJSON();

    let button: any = tree.children.find((el: any) => {
      return el.type === 'button';
    });

    let e = {
      preventDefault: () => {}
    };
    button.props.onClick(e);
    expect(pressed).toBe(1);
    
    button.props.onClick(e);
    expect(pressed).toBe(2);
  });

});
