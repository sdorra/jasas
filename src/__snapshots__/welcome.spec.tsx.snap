exports[`welcome should render text and button 1`] = `
<div
  className="welcome">
  <p>
    Welcome 
    tricia
  </p>
  <button
    onClick={[Function]}>
    Logout
  </button>
</div>
`;
